Feature/rpr 3 swagger api specification

gmtcrary opened the pull request.

Commit:
Add additional feature statements about API.

Adds feature statments for the API implementation project.

Commit:
Add note about API operations that need API key

Add note about API operations that require user authentication
needing an API key.
This commit is also a git --patch exercise.

brandonlee503 committed:
First commit, still things to change

brandonlee503 committed:
Rename TestSwaggerAPI to TestSwaggerAPI.yaml

brandonlee503 committed:
A few updates

Still plenty of bugs lol

brandonlee503 committed:
Fixed bugs so now it's only warnings lolol

brandonlee503 committed:
Multiple changes, see description.

Fixed warning messages, and added/modified more resource responses. Fixed "required" status for get requests.

brandonlee503 committed:
Updated security schemas with API keys

brandonlee503 committed:
Updated to fix errors noted in previous comments

See previous comments for more detail

brandonlee503 committed:
Updated to incorporate Challonge parameters

In tournament post.  However 3 errors appear with this.  They can be squelched when change the location to a single body rather than 3 queries.  I'll finish up this tomorrow, but see if you'd like to take a look first.

Commit:
Add security definitions and move challonge api key to query

Add security definitions and move challonge api key to query

Commit:
Apply security requirements to tournament post resource

Apply security requirements to tournament post resource

Commit:
Update tournament post parameters and security definition. Remove schema

Update tournament post parameters and security definition. Remove schema

Commit:
initial commit

Commit:
install gradle wrapper

per https://docs.gradle.org/current/userguide/gradle_wrapper.html

Commit:
initial gradle build script

Commit:
add to readme

Commit:
store groovy version in properties file

Commit:
add dropwizard dependency

Commit:
create application class and test

Commit:
update .gitignore

Commit:
create codenarc groovy ruleset

Commit:
lowercase package name

Commit:
document build task

Commit:
build depends on shadow jar

Commit:
use gradle distribution with sources for documentation

Commit:
merge service paths for shadow jar

Commit:
document build and run in readme

Commit:
initial configuration

Commit:
ignore osx metadata

Commit:
add minimal configuration

Commit:
add sample resource

Commit:
document sample resource

Commit:
add run task

Commit:
Change get user method parameters and description to match filtering purpose

Change get user method parameters and description to match filtering purpose

Commit:
Remove unneccesary api_key authentication on get user method

Remove unneccesary api_key authentication on get user method

Commit:
Add first draft of pseudocode

Add first draft of pseudocode

Commit:
Add headers to methods in psuedocode markdown

Add headers to methods in psuedocode markdown

Commit:
Add code block formating to methods

Add code block formating to methods

Commit:
Add markdown code block to PLAYERS/ELO pseudocode documentation

Add markdown code block to PLAYERS/ELO pseudocode documentation

Commit:
Move comments outside of code blocks

Move comments outside of code blocks

Commit:
Add dropwizard skeleton to project

Add dropwizard skeleton to project

Commit:
Add dropwizard user representation

Add dropwizard user representation

Commit:
Add UserDao and UserResource

Add UserDao and UserResource

Commit:
add ascii art

Commit:
Add updates to pseudocode, notes on DAO design, misc notes and TODOS RPR-9

Add updates to pseudocode, notes on DAO design, misc notes and TODOS RPR-9

Commit:
gnu agpl v3

Commit:
Add basic get post and byID get methods for User

Add basic get post and byID get methods for User

Commit:
Add ASCII Jigglypuff RESTPR banner

Add ASCII Jigglypuff RESTPR banner

Commit:
Fixed incorrect jdbi User POJO mapping

Fixed incorrect jdbi User POJO mapping

Commit:
todo: update jar name with version

Commit:
Update .gitignore and cleanup unneccesary build files

Update .gitignore and cleanup unneccesary build files

Commit:
Add new runscript to build.gradle and oracle driver dependancy

Add new runscript to build.gradle and oracle driver dependancy

brandonlee503 committed:
Update Readme.md for CAT-29

Performed the following updates:

1) clone the skeleton 
2) rename its branches so they don't conflict 
3) create master, develop, and feature branches 
4) backup existing code from your repo 

Haven't documented these yet: 
5) updating the remote repo 
6) requesting pulls from your repo to the skeleton

Commit:
Remove 404 error code from Specification User/get method

Remove 404 error code from Specification User/get method

Commit:
Add ErrorPOJO to restpr.api package

Add ErrorPOJO to restpr.api package

Commit:
Update specification POST /user method body parameter description

Update specification POST /user method body parameter description

Commit:
Remove 200, 202 response from POST /User and update 201 schema. Add urilocation

Remove 200, 202 response from POST /User and update 201 schema. Add urilocation

Commit:
Add GET /User/all method to specification

Add GET /User/all method to specification

Commit:
Change POST /User 405 response to 409 and description

Change POST /User 405 response to 409 and description

Commit:
Change PUT /User/{user_id} 202 response to 201

Change PUT /User/{user_id} 202 response to 201

Commit:
Add manualjavacall.sh manual java runscript for dropwizard app

Add manualjavacall.sh manual java runscript for dropwizard app

Commit:
Update gradle.properties mainClass property for dropwizard app

Update gradle.properties mainClass property for dropwizard app

Commit:
Add name to User pojo @XmlRootElement annotation

Add name to User pojo @XmlRootElement annotation

Commit:
Add Like clause matching to getPRUSERSmatch method in UserDAO

Add Like clause matching to getPRUSERSmatch method in UserDAO

Commit:
Change allRESTPRUsers SQL Select collumns from * to specific User pojo fields

Change allRESTPRUsers SQL Select collumns from * to specific User pojo fields

Commit:
Bug fix postUser SQL Statement to bind values correctly to collumn insert order

Bug fix postUser SQL Statement to bind values correctly to collumn insert order

Commit:
Add proper SQL Update statement to UserDAO putUser method

Add proper SQL Update statement to UserDAO putUser method

Commit:
Add postUserToUserId method and SQL statement to UserDAO

Add postUserToUserId method and SQL statement to UserDAO

Commit:
Add deleteUserById method and SQL statement to UserDAO

Add deleteUserById method and SQL statement to UserDAO

Commit:
Add getLatestUserId method and SQL statement to UserDAO

Add getLatestUserId method and SQL statement to UserDAO

Commit:
Add new restpr.api POJO imports and remove uneccesary imports

Add new restpr.api POJO imports and remove uneccesary imports

Commit:
Add debugPrintUser method to UserResource.groovy

Add debugPrintUser method to UserResource.groovy

Commit:
Update GET User/all method to return Jax RS response codes to spec instead of raw JSON

Update User/all resource method to return Jax RS response codes to spec instead of raw JSON

Commit:
Add user_login parameter to GET USER/ method and update to return Jax RS response codes to spec instead of raw JSON

Add user_login parameter to GET USER/ method and update to return Jax RS response codes to spec instead of raw JSON

Commit:
Update POST User/ method to return Jax RS response codes (201 (lacking URI), 409, 500) instead of just raw JSON

Update POST User/ method to return Jax RS response codes (201 (lacking URI), 409, 500) instead of just raw JSON

Commit:
Update GET User/{user_id} method to return Jax RS response codes to spec instead of raw JSON

Update GET User/{user_id} method to return Jax RS response codes to spec instead of raw JSON

Commit:
Add PUT User/{user_id} method that returns Jax RS response codes (200, 201, 404,500)

Add PUT User/{user_id} method that returns Jax RS response codes (200, 201, 404,500)

Commit:
Add DELETE User/{user_id} method that returns Jax RS response codes (200,500)

Add DELETE User/{user_id} method that returns Jax RS response codes (200,500)

Commit:
Cleanup old tracked tmp file and add .gradle/ to gitignore file

Cleanup old tracked tmp file and add .gradle/ to gitignore file

Commit:
Update idea configuration files to latest build

Update idea configuration files to latest build

Commit:
Remove invalid response codes from PUT User/{user_id} in the specification

Remove invalid response codes from PUT User/{user_id} in the specification

Commit:
Merge branch 'skeleton-master' into feature/RPR-3-Swagger-API-Specification

Conflicts:
	.gitignore
	build.gradle
	config/codenarc/codenarc.groovy
	gradle.properties
	gradle/wrapper/gradle-wrapper.jar
	gradle/wrapper/gradle-wrapper.properties
	gradlew.bat
	src/main/resources/banner.txt

Commit:
space code blocks consistently

Commit:
update section title

Commit:
remove unnecessary line

Commit:
update merge documentation

Commit:
Merge branch 'feature/cat-29-readme-commands'

ghost commented:
stop tracking all `.DS_Store` files

ghost commented:
In your readme, document examples of API calls using `netcat`. Talk with @brandonlee503 about using I/O redirection for that command.

ghost commented:
In your readme, document the steps you take to configure, build, and run the application.

ghost commented:
In the readme, remove references to unimplemented endpoints. The readme should reflect the current state of the code.

ghost commented:
in the readme, consider linking to your swagger file and your pseudocode.

ghost commented:
if you don't plan on using the pseudocode again, remove it.

ghost commented:
the license in the skeleton is the GNU AGPL v3. update your swagger file to reflect this.

ghost commented:
in your swagger file, remove `application/xml` return types that you do not plan to implement

ghost commented:
remove `src/test` if you do not plan on implementing unit tests for this pull request

ghost commented:
convert `RESTPRConfiguration.java` to groovy

ghost commented:
remove `SampleResource.groovy`

ghost commented:
general comment: i recommend defining explicit return types and avoiding groovy `def` statements where possible. this facilitates debugging and makes code more self-documenting.

ghost commented:
remove debugging statements that you don't want merged

ghost commented:
add javadoc comments to document public methods

ghost commented:
remove `Sample.groovy`

ghost commented:
use implicit getters and setters in `ErrorPOJO.groovy` and `User.groovy`

Commit:
codenarc extraneous semicolon warning

ghost commented:
incorporate the most recent change from the web api skeleton and solve the errors it causes

Commit:
Merge branch 'skeleton-master' into feature/RPR-3-Swagger-API-Specification

Merge branch 'skeleton-master' into feature/RPR-3-Swagger-API-Specification
This merge pulls in the new Codenarc rule of UnneccesarySemicolons

Commit:
Cleanup of idea files

Cleanup of idea files

Commit:
Update source files to comply with codenarc UnnecessarySemicolon rule

Update source files to comply with codenarc UnnecessarySemicolon rule

baldrugger commented:
I agree.  I noticed that we have some of this syntax:

Method signature: public Response getByCrn(@PathParam('crn') IntParam crn)

Variable to be returned: def returnResponse

Given that this is the variable returned by the function, it should be declared with an explicit datatype, Response.  In some cases in groovy, the MVC architecture imposes the use of inexplicit return types.  In all other cases, explicitly declare datatypes.

Good work, you all!  Keep it up. ☺

Cheers,
Mark
From: Taylor Brown [mailto:notifications@github.com]
Sent: Wednesday, July 15, 2015 3:10 PM
To: osu-mist/RestPR
Subject: Re: [RestPR] Feature/rpr 3 swagger api specification (#2)

general comment: i recommend defining explicit return types and avoiding groovy def statements where possible. this facilitates debugging and makes code more clear.

—
Reply to this email directly or view it on GitHubhttps://github.com/osu-mist/RestPR/pull/2#issuecomment-121764541.

Commit:
document example database configuration

Commit:
document usage with netcat

Commit:
remove spaces between imports

Commit:
remove spaces after braces

Commit:
unnecessary return keyword warning

Commit:
warn missing braces

Commit:
do not require braces for else if

browtayl committed:
Merge pull request #5 from osu-mist/feature/co-172-return

unnecessary return keyword warning

browtayl committed:
Merge pull request #6 from osu-mist/feature/co-173-braces

Feature/co 173 braces

Commit:
Cleanup and untrack build temporary files

Cleanup and untrack build temporary files

Commit:
Remove duplicate of manualjavacall.sh

Remove duplicate of manualjavacall.sh

Commit:
Reorder imports to Taylor's suggestions

Reorder imports to Taylor's suggestions.
https://github.com/osu-mist/RestPR/commit/5ca312747aff63902d2663ab0d43af6ab06b9c80#commitcomment-12187711

Commit:
Remove unneccesarry import from RESTPRApplication.groovy

Remove unneccesarry import from RESTPRApplication.groovy

Commit:
Remove explicit getters and setters

Remove explicit getters and setters. https://github.com/osu-mist/RestPR/commit/5ca312747aff63902d2663ab0d43af6ab06b9c80#commitcomment-12187825

Commit:
Add bash crunchbang to manualjavacall.sh

Add bash crunchbang to manualjavacall.sh

Commit:
Untrack unneccesarry .DS_Store files

Untrack unneccesarry .DS_Store files

Commit:
Remove references to unimplemented resources within README.md

Remove references to unimplemented resources within README.md

Commit:
Fix gradle.properties mainClass value to correct value

Fix gradle.properties mainClass value to correct value

Commit:
Remove src/test

Remove src/test. No unit tests are being implemented for this pull request.

Commit:
Convert RESTPRConfiguration.java to groovy

Convert RESTPRConfiguration.java to groovy

Commit:
Delete and untrack psuedocode.md

Delete and untrack psuedocode.md. Its no longers needed.

Commit:
Change name of TestSwaggerAPI.yaml to RestPRApi.yaml

Change name of TestSwaggerAPI.yaml to RestPRApi.yaml

Commit:
Add reference to RestPRApi.yaml file.

Add reference to RestPRApi.yaml file.

Commit:
Change license specified in RestPRApi.yaml to GNU AGPL v3

Change license specified in RestPRApi.yaml to GNU AGPL v3

Commit:
Remove application/xml production types from Swagger specification

Remove application/xml production types from Swagger specification

Commit:
Delete and untrack SampleResource.groovy

Delete and untrack SampleResource.groovy

Commit:
Delete and untrack Sample.groovy

Delete and untrack Sample.groovy

Commit:
Remove spaces between imports

Remove spaces between imports

Commit:
Remove final references to SampleResource

Remove final references to SampleResource

Commit:
Remove unneccesary debug statements and functions

Remove unneccesary debug statements and functions

Commit:
Remove additional unnecessary debug statement

Remove additional unnecessary debug statement

Commit:
Add javadoc comments to public functions in UserDAO and UserResource

Add javadoc comments to public functions in UserDAO and UserResource

Commit:
Fix variable declarations to explicit types

Fix variable declarations to explicit types

Commit:
Remove unneccesary access modifiers in RESTPRConfiguration.groovy

Remove unneccesary access modifiers in RESTPRConfiguration.groovy

Commit:
Add documentation to README.md about configuration, build and run steps

Add documentation to README.md about configuration, build and run steps

Commit:
Add header to Swagger reference in README.md

Add header to Swagger reference in README.md

Commit:
Add connect examples to README.md

Add connect examples to README.md

Commit:
Merge branch 'skeleton-master' into feature/RPR-3-Swagger-API-Specification

Merge branch 'skeleton-master' into feature/RPR-3-Swagger-API-Specification

Commit:
Fix source files to adhere to UnnecessaryReturnKeyword rule

Fix source files to adhere to UnnecessaryReturnKeyword rule

Commit:
Add created uri header response and bugfix usage of getters/setters

Add created uri header response and bugfix usage of getters/setters

Commit:
Add post and put netcat examples to README.md

Add post and put netcat examples to README.md

Commit:
Remove dollar signs from README.md markdown

Remove dollar signs from README.md markdown

browtayl commented on README.md:
```diff
@@ -2,34 +2,138 @@
+
+#Connection Examples
+The following examples can be excuted with netcat using heredocs like this:
+	nc localhost 8008 << HERE
```
i'm not sure why github is formatting some of your code blocks and not others, but you can explicitly define code blocks in markdown by indenting 4 spaces or by surrounding code with triple backticks

browtayl commented on src/main/groovy/edu/oregonstate/mist/restpr/RESTPRConfiguration.groovy:
```diff
@@ -0,0 +1,76 @@
+/**
+ * Created by georgecrary on 6/29/15.
+ */
+public class RESTPRConfiguration extends Configuration {
```
i'm not sure it's necessary to have so much stuff in your configuration class.

in my sad api i was able to pare this down to 8 lines:
 https://github.com/osu-mist/sad-api/blob/d8add8c0c24fa0e08168ab9c8cb5d7370a31c4b9/src/main/groovy/edu/oregonstate/mist/sadapi/SadApplicationConfiguration.groovy

browtayl commented on src/main/groovy/edu/oregonstate/mist/restpr/api/User.groovy:
```diff
@@ -0,0 +1,26 @@
+ * Created by georgecrary on 6/26/15.
+ */
+
+@XmlRootElement(name = "user")
```
if the `@XmlRootElement` annotation is necessary, explain why. otherwise delete it.

browtayl commented on src/main/groovy/edu/oregonstate/mist/restpr/db/UserDAO.groovy:
```diff
@@ -0,0 +1,103 @@
+  @SqlQuery("""
+            SELECT PR_USER_SEQ.CURRVAL FROM DUAL
+            """)
+  Integer getLatestUserId()
```
did you resolve the question about potential race conditions? if not, consider adding a comment like `FIXME: possible race condition`

gmtcrary replied:
This was where my concerns for transactions with JDBI started. For now I'll leave a fixme.

Commit:
Add hyperlink to RestPRApi.yaml in README.md

Add hyperlink to RestPRApi.yaml in README.md

Commit:
Remove unneccesary configuration statement in RESTPRConfiguration.groovy

Remove unneccesary configuration statement in RESTPRConfiguration.groovy

Commit:
Remove unnecessary XmlRootElement annotation and imports

Remove unnecessary XmlRootElement annotation and imports

Commit:
Add FIXME note to deleteUserById method in UserDAO

Add FIXME note to deleteUserById method in UserDAO
